pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +539,95 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_for_diag_eye_mul(fgraph, node):
+    """
+     This rewrite takes advantage of the fact that for a diagonal matrix, the inverse is a diagonal matrix with the new diagonal entries as reciprocals of the original diagonal elements.
+     This function deals with diagonal matrix arising from the multiplicaton of eye with a scalar/vector/matrix
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        Function graph being optimized
+    node: Apply
+        Node of the function graph to be optimized
+
+    Returns
+    -------
+    list of Variable, optional
+        List of optimized variables, or None if no optimization was performed
+    """
+    # List of useful operations : Inv, Pinv
+    valid_inverses = (MatrixInverse, MatrixPinv)
+    core_op = node.op.core_op
+    if not (isinstance(core_op, valid_inverses)):
+        return None
+
+    # Dealing with diagonal matrix from eye_mul
+    potential_mul_input = node.inputs[0]
+    eye_non_eye_inputs = _find_diag_from_eye_mul(potential_mul_input)
+    if eye_non_eye_inputs is not None:
+        eye_input, non_eye_inputs = eye_non_eye_inputs
+    else:
+        return None
+
+    # Dealing with only one other input
+    if len(non_eye_inputs) != 1:
+        return None
+
+    useful_eye, useful_non_eye = eye_input[0], non_eye_inputs[0]
+
+    # For a matrix, we can first get the diagonal and then only use those
+    if useful_non_eye.type.broadcastable[-2:] == (False, False):
+        # For Matrix
+        return [useful_eye * 1 / useful_non_eye.diagonal(axis1=-1, axis2=-2)]
+    else:
+        # For Scalar/Vector
+        return [useful_eye * 1 / useful_non_eye]
+
+
+def rewrite_inv_for_diag_ptdiag(fgraph, node):
+    pass
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_for_orthonormal(fgraph, node):
+    """
+     This rewrite takes advantage of the fact that for an orthonormal matrix, the inverse is simply the transpose.
+     This function deals with orthonormal matrix arising from pt.linalg.svd decomposition (U, Vh) or arising from pt.linalg.qr
```

Comment:
It doesn't look like you're dealing with the outputs of QR yet?

Reply:
yea i still have to add this